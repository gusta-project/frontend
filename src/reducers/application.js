import { buildActions } from 'utils';

export const types = buildActions('application', [
  'INIT_APP',
  'LOGIN_USER',
  'REQUEST_TOKEN',
  'RECEIVE_TOKEN',
  'REQUEST_CURRENT_USER',
  'RECEIVE_CURRENT_USER',
  'LOGIN_USER_SUCCESS',
  'LOGIN_USER_FAILURE',
  'LOGOUT_USER',
  'LOGOUT_USER_SUCCESS',
  'LOGOUT_USER_FAILURE',
  'REGISTER_USER',
  'REGISTER_USER_SUCCESS',
  'REGISTER_USER_FAILURE'
]);

const initApp = () => ({
  type: types.INIT_APP
});

const loginUser = (emailAddress, password) => ({
  type: types.LOGIN_USER,
  emailAddress,
  password
});

const requestToken = (emailAddress, password) => ({
  type: types.REQUEST_TOKEN,
  emailAddress,
  password
});

const receiveToken = (token, expiration) => ({
  type: types.RECEIVE_TOKEN,
  expiration,
  token
});

const requestCurrentUser = () => ({
  type: types.REQUEST_CURRENT_USER
});

const receiveCurrentUser = user => ({
  type: types.RECEIVE_CURRENT_USER,
  user
});

const loginUserSuccess = user => ({
  type: types.LOGIN_USER_SUCCESS,
  user
});

const loginUserFailure = error => ({
  type: types.LOGIN_USER_FAILURE,
  error
});

const logoutUser = () => ({
  type: types.LOGOUT_USER
});

const logoutUserSuccess = () => ({
  type: types.LOGOUT_USER_SUCCESS
});

const logoutUserFailure = error => ({
  type: types.LOGOUT_USER_FAILURE,
  error
});

const registerUser = details => ({
  type: types.REGISTER_USER,
  details
});

const registerUserSuccess = () => ({
  type: types.REGISTER_USER_SUCCESS
});

const registerUserFailure = error => ({
  type: types.REGISTER_USER_FAILURE,
  error
});

export const actions = {
  initApp,
  loginUser,
  requestToken,
  receiveToken,
  requestCurrentUser,
  receiveCurrentUser,
  loginUserSuccess,
  loginUserFailure,
  logoutUser,
  logoutUserSuccess,
  logoutUserFailure,
  registerUser,
  registerUserSuccess,
  registerUserFailure
};

export const initialState = {
  loggingIn: false,
  loggingOut: false,
  user: null,
  error: null,
  authorization: {
    accessToken: null,
    refreshToken: null,
    expiration: null
  },
  registration: {
    registering: false,
    complete: false,
    error: null
  }
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.LOGIN_USER:
      return {
        ...state,
        loggingIn: true
      };
    case types.LOGOUT_USER:
      return {
        ...state,
        loggingOut: true
      };
    case types.REGISTER_USER:
      return {
        ...state,
        registration: {
          ...state.registration,
          registering: true,
          details: action.details
        }
      };
    case types.RECEIVE_TOKEN:
      return {
        ...state,
        authorization: {
          accessToken: action.token,
          expiration: action.expiration
        }
      };
    case types.RECEIVE_CURRENT_USER:
      return {
        ...state,
        user: action.user
      };
    case types.LOGOUT_USER_SUCCESS:
      return {
        ...state,
        loggingOut: false,
        user: null,
        authorization: initialState.authorization
      };
    case types.LOGIN_USER_SUCCESS:
      return {
        ...state,
        loggingIn: false
      };
    case types.LOGIN_USER_FAILURE:
    case types.LOGOUT_USER_FAILURE:
      return {
        ...state,
        loggingIn: false,
        loggingOut: false,
        error: action.error
      };
    case types.REGISTER_USER_SUCCESS:
      return {
        ...state,
        registration: {
          registering: false,
          complete: true,
          error: null
        }
      };
    case types.REGISTER_USER_FAILURE:
      return {
        ...state,
        registration: {
          registering: false,
          complete: true,
          error: action.error
        }
      };
    default:
      return state;
  }
};
