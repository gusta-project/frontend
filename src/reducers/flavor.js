import { buildActions } from 'utils';

export const types = buildActions('flavor', [
  'REQUEST_STASH',
  'REQUEST_STASH_SUCCESS',
  'REQUEST_STASH_FAILURE'
]);

const requestStash = () => ({
  type: types.REQUEST_STASH
});

const requestStashSuccess = stash => ({
  type: types.REQUEST_STASH_SUCCESS,
  stash
});

const requestStashFailure = error => ({
  type: types.REQUEST_STASH_FAILURE,
  error
});

export const actions = {
  requestStash,
  requestStashSuccess,
  requestStashFailure
};

export const initialState = {
  loaded: false,
  error: null,
  stash: []
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_STASH_SUCCESS:
      return {
        ...state,
        stash: action.stash,
        loaded: true
      };
    case types.REQUEST_STASH_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
