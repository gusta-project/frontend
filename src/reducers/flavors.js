import { buildActions } from 'utils';

export const types = buildActions('flavors', [
  'REQUEST_FLAVORS',
  'REQUEST_FLAVORS_SUCCESS',
  'REQUEST_FLAVORS_FAILURE'
]);

const requestFlavors = (pager) => ({
  type: types.REQUEST_FLAVORS,
  pager
});

const requestFlavorsSuccess = (flavors, pager) => ({
  type: types.REQUEST_FLAVORS_SUCCESS,
  flavors,
  pager
});

const requestFlavorsFailure = (error) => ({
  type: types.REQUEST_FLAVORS_FAILURE,
  error
});

export const actions = {
  requestFlavors,
  requestFlavorsSuccess,
  requestFlavorsFailure
};

export const initialState = {
  cache: [],
  collection: [],
  pager: {
    count: null,
    limit: 100,
    page: 1,
    pages: null
  }
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_FLAVORS_SUCCESS:
      return {
        ...state,
        cache: action.flavors,
        collection: action.flavors[action.pager.page],
        pager: {
          ...state.pager,
          ...action.pager
        }
      };
    case types.REQUEST_FLAVORS_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
