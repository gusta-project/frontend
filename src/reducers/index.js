import { combineReducers } from 'redux';

import {
  reducer as application,
  initialState as applicationState
} from './application';
import { reducer as profile, initialState as profileState } from './profile';
import { reducer as toast, initialState as toastState } from './toast';

export const initialState = {
  application: applicationState,
  profile: profileState,
  toast: toastState
};

export default combineReducers({
  application,
  profile,
  toast
});
