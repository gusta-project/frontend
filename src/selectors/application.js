import dayjs from 'dayjs';
import { createSelector } from 'reselect';

export const getApplication = state => state.application;

export const isLoggingIn = createSelector(
  getApplication,
  application => Boolean(application.loggingIn)
);

export const isLoggingOut = createSelector(
  getApplication,
  application => Boolean(application.loggingOut)
);

export const getUser = createSelector(
  getApplication,
  application => application.user
);

export const getError = createSelector(
  getApplication,
  application => application.error
);

export const getAuthorization = createSelector(
  getApplication,
  application => application.authorization
);

export const getToken = createSelector(
  getAuthorization,
  authorization => authorization.accessToken
);

export const getTokenExpiration = createSelector(
  getAuthorization,
  authorization => authorization.expiration
);

export const isLoggedIn = createSelector(
  [getToken, getTokenExpiration],
  (token, expiration) => Boolean(token && dayjs().isBefore(dayjs(expiration)))
);

export const getRegistration = createSelector(
  getApplication,
  application => application.registration
);

export const isRegistering = createSelector(
  getRegistration,
  registration => registration.registering && !registration.complete
);
