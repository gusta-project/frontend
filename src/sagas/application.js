import dayjs from 'dayjs';
import { all, call, put, takeLatest, take, select } from 'redux-saga/effects';

import request from 'utils/request';
import { actions, types } from 'reducers/application';
import { actions as toastActions } from 'reducers/toast';
import { isLoggedIn, getUser } from 'selectors/application';

// snake_cased variables here come from RFC 6749
/* eslint-disable camelcase */
function* requestTokenWorker({ emailAddress, password }) {
  try {
    const endpoint = {
      url: '/oauth/token',
      method: 'POST'
    };
    const data = {
      grant_type: 'password',
      username: emailAddress,
      password
    };
    const result = yield call(request.execute, { endpoint, data });

    // update token info in state or throw an error
    if (result.success) {
      const {
        data: {
          access_token: token,
          token_type: tokenType,
          expires_in: expiresIn
        }
      } = result.response;

      if (tokenType !== 'Bearer') {
        throw new Error(`Unable to use token of type ${tokenType}`);
      }

      const expiration = dayjs().add(expiresIn, 'seconds');

      yield put(actions.requestTokenSuccess({ token, expiration }));
    } else if (result.error) {
      throw result.error;
    } else {
      throw new Error('Request failed for an unspecified reason!');
    }
  } catch (error) {
    const { message } = error;

    yield put(actions.requestTokenFailure(error));
    yield put(
      toastActions.popToast({
        title: 'Error!',
        icon: 'times-circle',
        message
      })
    );
  }
}
/* eslint-enable camelcase */

function* requestCurrentUserWorker() {
  try {
    const endpoint = {
      url: '/user/current',
      method: 'GET'
    };
    const result = yield call(request.execute, { endpoint });

    // update user in state or throw an error
    if (result.success) {
      const {
        response: { data }
      } = result;

      yield put(actions.requestCurrentUserSuccess(data));
    } else if (result.error) {
      throw result.error;
    } else {
      throw new Error('Failed to get current user!');
    }
  } catch (error) {
    const { message } = error;

    yield put(actions.requestCurrentUserFailure(error));
    yield put(
      toastActions.popToast({
        title: 'Error',
        icon: 'times-circle',
        message
      })
    );
  }
}

function* loginUserWorker({ emailAddress, password }) {
  try {
    // first, check to see if we are already logged in
    const loggedIn = yield select(isLoggedIn);

    if (!loggedIn) {
      // obtain a bearer token
      // then, obtain current user information
      // use take because it is blocking
      yield put(actions.requestToken(emailAddress, password));
      const tokenResult = yield take([
        types.REQUEST_TOKEN_SUCCESS,
        types.REQUEST_TOKEN_FAILURE
      ]);

      if (tokenResult.type === types.REQUEST_TOKEN_FAILURE) {
        throw new Error('Failed to log in!');
      }

      const { token, expiration } = tokenResult;

      localStorage.setItem('accessToken', JSON.stringify(token));
      localStorage.setItem(
        'expiration',
        JSON.stringify(expiration.toISOString())
      );
    }

    yield put(
      toastActions.popToast({
        title: 'Logged in',
        icon: 'check',
        message: 'You have been authenticated.'
      })
    );

    let user = yield select(getUser);

    if (!user) {
      yield put(actions.requestCurrentUser());
      const currentUserResult = yield take([
        types.REQUEST_CURRENT_USER_SUCCESS,
        types.REQUEST_CURRENT_USER_FAILURE
      ]);

      if (currentUserResult.type === types.REQUEST_CURRENT_USER_FAILURE) {
        throw new Error('Failed to fetch current user!');
      }

      user = currentUserResult.user;

      if (!user) {
        throw new Error('Got invalid response to current user request!');
      }
    }

    yield put(actions.loginUserSuccess());
  } catch (error) {
    const { message } = error;

    yield put(actions.loginUserFailure(error));
    yield put(
      toastActions.popToast({
        title: 'Error',
        icon: 'times-circle',
        message
      })
    );
  }
}

function* logoutUserWorker() {
  try {
    // first, check to see if we are logged in
    const loggedIn = yield select(isLoggedIn);

    if (loggedIn) {
      localStorage.removeItem('accessToken');
      localStorage.removeItem('expiration');
    }

    yield put(
      toastActions.popToast({
        title: 'Logged out',
        icon: 'check',
        message: 'You have been logged out.'
      })
    );

    yield put(actions.logoutUserSuccess());
  } catch (error) {
    const { message } = error;

    yield put(actions.logoutUserFailure(error));
    yield put(
      toastActions.popToast({
        title: 'Error',
        icon: 'times-circle',
        message
      })
    );
  }
}

function* registerUserWorker({
  details: { emailAddress, password, username }
}) {
  try {
    const endpoint = {
      url: '/register',
      method: 'POST'
    };
    const data = {
      emailAddress,
      password,
      username
    };
    const result = yield call(request.execute, { endpoint, data });

    if (result.success) {
      yield put(actions.registerUserSuccess());
      yield put(
        toastActions.popToast({
          title: 'Success!',
          icon: 'check',
          message: 'Check your email for an activation link.'
        })
      );
    } else {
      throw result.error;
    }
  } catch (error) {
    const { message } = error;

    yield put(actions.registerUserFailure(error));
    yield put(
      toastActions.popToast({
        title: 'Error',
        icon: 'times-circle',
        message
      })
    );
  }
}

function* loginUserWatcher() {
  yield takeLatest(types.LOGIN_USER, loginUserWorker);
}

function* logoutUserWatcher() {
  yield takeLatest(types.LOGOUT_USER, logoutUserWorker);
}

function* requestTokenWatcher() {
  yield takeLatest(types.REQUEST_TOKEN, requestTokenWorker);
}

function* requestCurrentUserWatcher() {
  yield takeLatest(types.REQUEST_CURRENT_USER, requestCurrentUserWorker);
}

function* registerUserWatcher() {
  yield takeLatest(types.REGISTER_USER, registerUserWorker);
}

export const workers = {
  loginUserWorker,
  logoutUserWorker,
  registerUserWorker,
  requestTokenWorker,
  requestCurrentUserWorker
};

export const watchers = {
  loginUserWatcher,
  logoutUserWatcher,
  registerUserWatcher,
  requestTokenWatcher,
  requestCurrentUserWatcher
};

export default function* saga() {
  yield all(Object.values(watchers).map(watcher => watcher()));
}
