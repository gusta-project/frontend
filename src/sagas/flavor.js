import { all, call, put, takeLatest, select, take } from 'redux-saga/effects';

import request from 'utils/request';
import { getUser } from 'selectors/application';
import { actions, types } from 'reducers/flavor';
import { actions as appActions, types as appTypes } from 'reducers/application';

function* requestStashWorker() {
  try {
    let user = yield select(getUser);

    if (user === null) {
      yield put(appActions.requestCurrentUser());
      yield take([
        appTypes.REQUEST_CURRENT_USER_SUCCESS,
        appTypes.REQUEST_CURRENT_USER_FAILURE
      ]);
      user = yield select(getUser);
    }

    const endpoint = {
      url: `/user/${user.id}/flavors`,
      method: 'GET'
    };
    const result = yield call(request.execute, { endpoint });

    if (result.success) {
      const { data } = result.response;

      if (!data) {
        return yield put(actions.requestStashSuccess([]));
      }

      yield put(actions.requestStashSuccess(data));
    } else if (result.error) {
      throw result.error;
    } else {
      throw new Error('Request failed for an unspecified reason!');
    }
  } catch (error) {
    // eslint-disable-next-line
    console.dir(error);
    yield put(actions.requestStashFailure(error));
  }
}

export const workers = {
  requestStashWorker
};

function* requestStashWatcher() {
  yield takeLatest(types.REQUEST_STASH, requestStashWorker);
}

export const watchers = {
  requestStashWatcher
};

export default function* saga() {
  yield all(Object.values(watchers).map(watcher => watcher()));
}
