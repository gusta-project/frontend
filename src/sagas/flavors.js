import { all, call, put, takeLatest, select } from 'redux-saga/effects';
import helper from 'utils/saga';
import { getCachedFlavors, getFlavorsPager } from 'selectors/flavors';
import { actions, types } from 'reducers/flavors';

function* requestFlavorsWorker({ pager }) {
  try {
    const cached = yield select(getCachedFlavors);
    const store = yield select(getFlavorsPager);

    const response = yield call(helper.pager, {
      cached,
      pager: {
        ...pager,
        store
      },
      route: {
        count: '/flavors/count',
        data: '/flavors/'
      },
      type: 'Flavors'
    });

    yield put(actions.requestFlavorsSuccess(response.cached, response.pager));
  } catch (error) {
    const { message } = error;

    yield put(actions.requestFlavorsFailure(error));
    yield call(helper.errorToast, message);
  }
}

function* requestFlavorsWatcher() {
  yield takeLatest(types.REQUEST_FLAVORS, requestFlavorsWorker);
}

export const workers = {
  requestFlavorsWorker
};

export const watchers = {
  requestFlavorsWatcher
};

export default function* saga() {
  yield all(Object.values(watchers).map(watcher => watcher()));
}
