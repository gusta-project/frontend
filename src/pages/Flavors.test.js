import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { initialState as appInitialState } from 'reducers/application';
import { initialState as flavorInitialState } from 'reducers/flavor';
import { initialState as flavorsInitialState } from 'reducers/flavors';
import ConnectedFlavors, { Flavors } from './Flavors';
import { withMemoryRouter } from 'utils/testing';

jest.mock('components/Pagination/Pagination', () => {
  const pagination = require('utils/testing').mockComponent('Pagination');
  const PagerInfo = require('utils/testing').mockComponent('PagerInfo');

  return {
    withPagination: () => () => pagination,
    pagination,
    PagerInfo
  };
});
jest.mock('components/ToggleButton/ToggleButton', () =>
  require('utils/testing').mockComponent('ToggleButton')
);

describe('Page <Flavors />', () => {
  const mockStore = configureStore();
  const actions = {
    requestStash: jest.fn(),
    requestFlavors: jest.fn(),
    addStash: jest.fn(),
    removeStash: jest.fn()
  };

  const collection = [
    {
      id: '1',
      vendorId: 3,
      name: '27 Bears',
      slug: null,
      density: null,
      Vendor: { id: 3, name: 'Capella', slug: 'capella', code: 'CAP' }
    },
    {
      id: '2',
      vendorId: 3,
      name: '27 Fish',
      slug: null,
      density: null,
      Vendor: { id: 3, name: 'Capella', slug: 'capella', code: 'CAP' }
    },
    {
      id: '3',
      vendorId: 3,
      name: 'Acai',
      slug: null,
      density: null,
      Vendor: { id: 3, name: 'Capella', slug: 'capella', code: 'CAP' }
    },
    {
      id: '4',
      vendorId: 3,
      name: 'Amaretto',
      slug: null,
      density: null,
      Vendor: { id: 3, name: 'Capella', slug: 'capella', code: 'CAP' }
    },
    {
      id: '5',
      vendorId: 3,
      name: 'Anise',
      slug: null,
      density: null,
      Vendor: { id: 3, name: 'Capella', slug: 'capella', code: 'CAP' }
    }
  ];
  const stash = [
    {
      userId: '1',
      flavorId: '1',
      created: '2020-03-02T12:10:33.787Z',
      minMillipercent: null,
      maxMillipercent: null,
      UserId: '1',
      Flavor: {
        id: '1',
        vendorId: 3,
        name: '27 Bears',
        slug: null,
        density: null,
        Vendor: { id: 3, name: 'Capella', slug: 'capella', code: 'CAP' }
      }
    },
    {
      userId: '1',
      flavorId: '2',
      created: '2020-03-02T12:10:34.135Z',
      minMillipercent: null,
      maxMillipercent: null,
      UserId: '1',
      Flavor: {
        id: '2',
        vendorId: 3,
        name: '27 Fish',
        slug: null,
        density: null,
        Vendor: { id: 3, name: 'Capella', slug: 'capella', code: 'CAP' }
      }
    },
    {
      userId: '1',
      flavorId: '3',
      created: '2020-03-02T12:10:34.641Z',
      minMillipercent: null,
      maxMillipercent: null,
      UserId: '1',
      Flavor: {
        id: '3',
        vendorId: 3,
        name: 'Acai',
        slug: null,
        density: null,
        Vendor: { id: 3, name: 'Capella', slug: 'capella', code: 'CAP' }
      }
    },
    {
      userId: '1',
      flavorId: '4',
      created: '2020-03-02T12:10:36.265Z',
      minMillipercent: null,
      maxMillipercent: null,
      UserId: '1',
      Flavor: {
        id: '4',
        vendorId: 3,
        name: 'Amaretto',
        slug: null,
        density: null,
        Vendor: { id: 3, name: 'Capella', slug: 'capella', code: 'CAP' }
      }
    }
  ];
  const pager = { count: 5, limit: 20, page: 1, pages: 1 };
  const store = mockStore({
    actions,
    application: appInitialState,
    flavor: flavorInitialState,
    flavors: flavorsInitialState
  });

  const pagerNavigation = <p></p>;

  const notLoggedInProps = {
    actions,
    collection,
    loggedIn: false,
    pager,
    pagerNavigation,
    stash,
    stashLoaded: false
  };

  const noStashProps = {
    actions,
    collection,
    loggedIn: true,
    pager,
    pagerNavigation,
    stash,
    stashLoaded: false
  };

  const props = {
    actions,
    collection,
    loggedIn: true,
    pager,
    pagerNavigation,
    stash,
    stashLoaded: true
  };

  const RoutedConnectedFlavors = withMemoryRouter(ConnectedFlavors);
  const RoutedFlavors = withMemoryRouter(Flavors);

  it('renders correctly', () => {
    expect(
      renderer
        .create(
          <Provider store={store}>
            <RoutedConnectedFlavors />
          </Provider>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it('renders correctly when stash is not loaded', () => {
    expect(
      renderer.create(<RoutedFlavors {...noStashProps} />).toJSON()
    ).toMatchSnapshot();
  });

  it('renders correctly when user is not logged in', () => {
    expect(
      renderer.create(<RoutedFlavors {...notLoggedInProps} />).toJSON()
    ).toMatchSnapshot();
  });

  const component = renderer.create(<RoutedFlavors {...props} />);

  const { instance } = component.root.findByType(Flavors);

  it('renders current user flavors correctly', () => {
    const tree = renderer.create(<RoutedFlavors {...props} />).toJSON;

    expect(tree).toMatchSnapshot();
  });

  it('can handle stashToggle', () => {
    const event = {
      target: { name: 'stashToggle', checked: true }
    };

    expect(instance.handleStashToggle).toBeDefined();
    instance.handleStashToggle(event);
    expect(instance.state.holdings).toEqual({
      1: true,
      2: true,
      3: true,
      4: true
    });
    expect(instance.state.stashToggle).toEqual(true);
  });

  it('can handle addToStash', () => {
    const id = 5;

    expect(instance.addToStash).toBeDefined();
    instance.addToStash(id);
    expect(actions.addStash).toBeCalledWith({ id });
    expect(instance.state.holdings).toEqual({
      1: true,
      2: true,
      3: true,
      4: true,
      5: true
    });
  });

  it('can handle removeFromStash', () => {
    const id = 5;

    expect(instance.removeFromStash).toBeDefined();
    instance.removeFromStash(id);
    expect(actions.removeStash).toBeCalledWith({ id });
    expect(instance.state.holdings).toEqual({
      1: true,
      2: true,
      3: true,
      4: true,
      5: false
    });
  });
});
