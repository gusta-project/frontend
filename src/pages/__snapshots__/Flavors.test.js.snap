// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page <Flavors /> renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="container-fluid"
  >
    <div
      className="text-center row"
    >
      <div
        className="col"
      >
        <h1>
          Flavors
        </h1>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="text-left ml-2 mb-4 col"
      >
        <small>
          Log in to Enable Flavor Stash
        </small>
      </div>
    </div>
    <div
      className="container-fluid"
    >
      <div
        className="pb-2 row"
      >
        <div
          className="col-3"
        >
          <input
            className="form-control"
            max="200"
            min="20"
            onBlur={[Function]}
            onChange={[Function]}
            step="20"
            type="number"
            value={20}
          />
        </div>
        <div
          className="text-right col"
        >
          <select
            onBlur={[Function]}
            onChange={[Function]}
            value={1}
          >
            <option
              value={1}
            >
              Page 
              1
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="table-responsive"
  >
    <table
      className="table table-sm table-striped table-bordered table-hover"
    >
      <thead>
        <tr
          className="text-center"
        >
          <th>
            ID
          </th>
          <th>
            Vendor
          </th>
          <th>
            Name
          </th>
          <th>
            Slug
          </th>
          <th>
            Density
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
  <div
    className="container-fluid"
  >
    <div
      className="pb-2 row"
    >
      <div
        className="col-3"
      >
        <input
          className="form-control"
          max="200"
          min="20"
          onBlur={[Function]}
          onChange={[Function]}
          step="20"
          type="number"
          value={20}
        />
      </div>
      <div
        className="text-right col"
      >
        <select
          onBlur={[Function]}
          onChange={[Function]}
          value={1}
        >
          <option
            value={1}
          >
            Page 
            1
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
         
        Flavors
      </div>
      <div
        className="text-right col"
      >
         Pages
      </div>
    </div>
  </div>
</div>
`;

exports[`Page <Flavors /> renders correctly when stash is not loaded 1`] = `
<div
  className="container"
>
  <div
    className="container-fluid"
  >
    <div
      className="text-center row"
    >
      <div
        className="col"
      >
        <h1>
          Flavors
        </h1>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="text-left ml-2 mb-4 col"
      >
        <ToggleButton />
         
        <span>
          Enable Flavor Stash
        </span>
      </div>
    </div>
    <p />
  </div>
  <div
    className="table-responsive"
  >
    <table
      className="table table-sm table-striped table-bordered table-hover"
    >
      <thead>
        <tr
          className="text-center"
        >
          <th>
            ID
          </th>
          <th>
            Vendor
          </th>
          <th>
            Name
          </th>
          <th>
            Slug
          </th>
          <th>
            Density
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="text-center"
          >
            1
          </td>
          <td>
            Capella
          </td>
          <td>
            27 Bears
          </td>
          <td
            className="text-center"
          />
          <td
            className="text-center"
          />
        </tr>
        <tr>
          <td
            className="text-center"
          >
            2
          </td>
          <td>
            Capella
          </td>
          <td>
            27 Fish
          </td>
          <td
            className="text-center"
          />
          <td
            className="text-center"
          />
        </tr>
        <tr>
          <td
            className="text-center"
          >
            3
          </td>
          <td>
            Capella
          </td>
          <td>
            Acai
          </td>
          <td
            className="text-center"
          />
          <td
            className="text-center"
          />
        </tr>
        <tr>
          <td
            className="text-center"
          >
            4
          </td>
          <td>
            Capella
          </td>
          <td>
            Amaretto
          </td>
          <td
            className="text-center"
          />
          <td
            className="text-center"
          />
        </tr>
        <tr>
          <td
            className="text-center"
          >
            5
          </td>
          <td>
            Capella
          </td>
          <td>
            Anise
          </td>
          <td
            className="text-center"
          />
          <td
            className="text-center"
          />
        </tr>
      </tbody>
    </table>
  </div>
  <p />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        5
         
        Flavors
      </div>
      <div
        className="text-right col"
      >
        1
         Pages
      </div>
    </div>
  </div>
</div>
`;

exports[`Page <Flavors /> renders correctly when user is not logged in 1`] = `
<div
  className="container"
>
  <div
    className="container-fluid"
  >
    <div
      className="text-center row"
    >
      <div
        className="col"
      >
        <h1>
          Flavors
        </h1>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="text-left ml-2 mb-4 col"
      >
        <small>
          Log in to Enable Flavor Stash
        </small>
      </div>
    </div>
    <p />
  </div>
  <div
    className="table-responsive"
  >
    <table
      className="table table-sm table-striped table-bordered table-hover"
    >
      <thead>
        <tr
          className="text-center"
        >
          <th>
            ID
          </th>
          <th>
            Vendor
          </th>
          <th>
            Name
          </th>
          <th>
            Slug
          </th>
          <th>
            Density
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="text-center"
          >
            1
          </td>
          <td>
            Capella
          </td>
          <td>
            27 Bears
          </td>
          <td
            className="text-center"
          />
          <td
            className="text-center"
          />
        </tr>
        <tr>
          <td
            className="text-center"
          >
            2
          </td>
          <td>
            Capella
          </td>
          <td>
            27 Fish
          </td>
          <td
            className="text-center"
          />
          <td
            className="text-center"
          />
        </tr>
        <tr>
          <td
            className="text-center"
          >
            3
          </td>
          <td>
            Capella
          </td>
          <td>
            Acai
          </td>
          <td
            className="text-center"
          />
          <td
            className="text-center"
          />
        </tr>
        <tr>
          <td
            className="text-center"
          >
            4
          </td>
          <td>
            Capella
          </td>
          <td>
            Amaretto
          </td>
          <td
            className="text-center"
          />
          <td
            className="text-center"
          />
        </tr>
        <tr>
          <td
            className="text-center"
          >
            5
          </td>
          <td>
            Capella
          </td>
          <td>
            Anise
          </td>
          <td
            className="text-center"
          />
          <td
            className="text-center"
          />
        </tr>
      </tbody>
    </table>
  </div>
  <p />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        5
         
        Flavors
      </div>
      <div
        className="text-right col"
      >
        1
         Pages
      </div>
    </div>
  </div>
</div>
`;

exports[`Page <Flavors /> renders current user flavors correctly 1`] = `[Function]`;
